### PR TITLE
Scripts for BMH CR creation in bulk

### PR DESCRIPTION
**Pull request by ParagDoke**
- Scripts for BMH CR creation in bulk
- Fix docker registry
- Capture BMO version instead of pf9-version

### CODE REVIEW

File: bulk_add/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+BUILD_NUMBER ?= 0
+BMH_VERSION ?= 0.5.1
```
Comment (poojaghumre) [inline bulk_add/Makefile]: rename this as METAL3_VERSION

File: bulk_add/ipmi_bulk.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (poojaghumre) [inline bulk_add/ipmi_bulk.sh]: Please rename this file as ipmi_address_bulk.sh

File: bulk_add/ipmi_bulk.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+bmc_user=$(echo ${USERNAME} | base64)
+bmc_pass=$(echo ${PASSWORD} | base64)
+port=${BMC_PORT:-6230}
+
+k8s_bearer_token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+curl_args=("--cacert" "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt" "-H" "Authorization: Bearer ${k8s_bearer_token}")
+
+# fetch addresses of bare metal hosts already known
+#IFS=','
```
Comment (poojaghumre) [inline bulk_add/ipmi_bulk.sh]: why is this line commented out?

File: bulk_add/port_bulk.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (poojaghumre) [inline bulk_add/port_bulk.sh]: Please rename this file as ipmi_port_bulk.sh

File: bulk_add/port_bulk.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ipmi_addresses=$(curl -s GET "${curl_args[@]}" https://kubernetes.default/apis/metal3.io/v1alpha1/baremetalhosts | jq '.items[].spec.bmc.address')
+
+for port in $(seq --separator=, ${VBMC_PORT_START} ${VBMC_PORT_END}); do
+    # we wish to ignore ports already known
+    # https://stackoverflow.com/a/15394738
+
+    if [[ ! "${ipmi_addresses[*]}:" =~ ":${port}" ]]; then
+        echo "No bare metal host present for IPMI port ${port} ..."
+        if ipmitool -I lanplus -U ${USERNAME} -P ${PASSWORD} -H ${ipmi_addr} -p ${port} power status; then
+			echo "Creating bmh test-pf9-${port}"
```
Comment (poojaghumre) [inline bulk_add/port_bulk.sh]: nit: the indentation seems to be a lot more than 4 spaces in this block?

File: bulk_add/ipmi_bulk.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ipmi_addresses=$(curl -s GET "${curl_args[@]}" https://kubernetes.default/apis/metal3.io/v1alpha1/baremetalhosts | jq '.items[].spec.bmc.address')
+
+# generate a new list of ip addresses in the range provided
+for addr in $(python3 -c "from ipaddress import IPv4Address as ip; print('\n'.join(str(ip(n)) for n in range(int(ip('${IPMI_RANGE_START}')), int(ip('${IPMI_RANGE_END}')) + 1)))"); do
+    # we wish to ignore addresses already known
+    # https://stackoverflow.com/a/15394738
+    if [[ ! "${ipmi_addresses[*]}:" =~ "${addr}:" ]]; then
+        echo "No bare metal host present for IPMI address ${addr} ..."
+        if ipmitool -I lanplus -U ${USERNAME} -P ${PASSWORD} -H ${addr} -p ${port} power status; then
+			echo "Creating bmh ${PREFIX}-${addr}"
```
Comment (poojaghumre) [inline bulk_add/ipmi_bulk.sh]: nit: the indentation seems to be a lot more than 4 spaces in this block?

**Review by poojaghumre — CHANGES_REQUESTED** [review]
Please squash the commits into one before merging

File: bulk_add/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,8 @@
+ARG BASE_IMAGE=quay.io/centos/centos:stream8
+FROM ${BASE_IMAGE} AS ironic-builder 
+RUN dnf install -y \
+    OpenIPMI \
+    ipmitool \
+    jq \
+    python3
+COPY ipmi_bulk.sh port_bulk.sh /bin/
```
Comment (poojaghumre) [inline bulk_add/Dockerfile]: you will need to change file names here too

File: bulk_add/ipmi_address_bulk.sh [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+ipmi_addresses=$(curl -s GET "${curl_args[@]}" https://kubernetes.default/apis/metal3.io/v1alpha1/baremetalhosts | jq '.items[].spec.bmc.address')
+
+# generate a new list of ip addresses in the range provided
+for addr in $(python3 -c "from ipaddress import IPv4Address as ip; print('\n'.join(str(ip(n)) for n in range(int(ip('${IPMI_RANGE_START}')), int(ip('${IPMI_RANGE_END}')) + 1)))"); do
+    # we wish to ignore addresses already known
+    # https://stackoverflow.com/a/15394738
+    if [[ ! "${ipmi_addresses[*]}:" =~ "${addr}:" ]]; then
+        echo "No bare metal host present for IPMI address ${addr} ..."
+        if ipmitool -I lanplus -U ${USERNAME} -P ${PASSWORD} -H ${addr} -p ${port} power status; then
+			echo "Creating bmh ${PREFIX}-${addr}"
```
Comment (poojaghumre) [inline bulk_add/ipmi_address_bulk.sh]: something is off with indentation still, below file too.. are you using tabs instead of spaces?

**Review by poojaghumre — CHANGES_REQUESTED** [review]

File: bulk_add/ipmi_bulk.sh (hunk 3 above)
Comment (shweta50) [reply bulk_add/ipmi_bulk.sh]: To solve iteration errors.

**Review by mayuresh108 — APPROVED** [review]

**Pull request merged**